bin/all_sky_search/pycbc_upload_single_event_to_gracedb

```diff
@@ -0,0 +1,195 @@
+                         "gracedb query to further filter events. "
+                         "Default='pycbc'")
+parser.add_argument('--search-id-string', default='AllSky',
+                    help="Using T050017-v1 naming convention for "
+                         "output XML filename. This is a string for "
+                         "search identifier in filename, e.g. 'AllSky' "
+                         "would give H1L1V1-PYCBC_AllSky-1234567890-1.xml. "
+                         "See https://dcc.ligo.org/LIGO-T050017/public. "
+                         "Default: 'AllSky'")
+parser.add_argument('--snr-timeseries-plot',
+                    help="SNR timeseries plot to be uploaded with the event.")
+parser.add_argument('--asd-plot',
+                    help="ASD plot to be uploaded with the event.")
+parser.add_argument('--skymap-plot',
+                    help="Skymap plot to be uploaded with the event.")
+parser.add_argument('--skymap-fits-file',
+                    help="Skymap .fits file to be uploaded with the event.")
+parser.add_argument('--source-probabilities',
+                    help="Source probabilities file to be uploaded with the "
+                         "event.")
+parser.add_argument('--source-probabilities-plot',
+                    help="Source probabilities plot to be uploaded with the "
+                          "event.")
+parser.add_argument('--labels', nargs='+',
+                    help="Labels to add to the event in GraceDB")
+
+args = parser.parse_args()
+
+pycbc.init_logging(logging.INFO)
+# Make the scitokens logger a little quieter
+# (it is called through GraceDB)
+logging.getLogger('scitokens').setLevel(logging.root.level + 10)
+
+if args.production_server:
+    gracedb = GraceDb()
+else:
+    gracedb = GraceDb(service_url='https://gracedb-playground.ligo.org/api/')
+
+labels = [l.upper() for l in args.labels]
+allowed_labels = gracedb.allowed_labels
+
+if set(labels) - set(allowed_labels):
+    err_msg = "One or more supplied labels is not available on the server. "
+    err_msg += f"Supplied {','.join(labels)}, allowed "
+    err_msg += f"{','.join(allowed_labels)}."
+    raise RuntimeError(err_msg)
+
+group_tag = 'Test' if args.testing else 'CBC'
+r = gracedb.create_event(
+    group_tag,
+    'pycbc',
+    args.xml_file_for_upload,
+    filecontents=open(args.xml_file_for_upload, "rb").read(),
+    search=args.search_id_string,
+    offline=True,
+    labels=labels
+).json()
+
+logging.info("Uploaded event %s.", r["graceid"])
+
+# add info for tracking code version
+gracedb_tag_with_version(gracedb, r['graceid'])
+
+# document the absolute path to the input file
+input_file_str = 'Candidate uploaded from ' \
+    + os.path.abspath(args.xml_file_for_upload)
+gracedb.write_log(r['graceid'], input_file_str)
+
+# document the command line used in the event log
+command_str = 'Upload command: ' \
+    + ' '.join(sys.argv)
+gracedb.write_log(r['graceid'], command_str)
```

Comment:
Maybe we also want the `analyst_comments` tag here. And in the spirit of minimizing the calls to GraceDB, is it better to collate all strings into a single longer comment?